pkg/karmadactl/promote/promote.go

```diff
@@ -114,6 +114,14 @@ type CommandPromoteOption struct {
 	// DryRun tells if run the command in dry-run mode, without making any server requests.
 	DryRun bool
 
+	// AutoCreatePropagationPolicy tells if the promote command create a PropagationPolicy automatically, otherwise one can create their own PropagationPolicy
+	AutoCreatePropagationPolicy bool
+
+	// PropagationPolicyName is the name of a PropagationPolicy that automatically created by promote command,
+	// if not specified, a suffix will be added to the PropagationPolicy.
+	// It is only valid when AutoCreatePropagationPolicy is true.
+	PropagationPolicyName string
```

Comment:
```suggestion
	// AutoCreatePolicy tells if the promote command should create the
	// PropagationPolicy(or ClusterPropagationPolicy) by default.
	// It defaults to true.
	AutoCreatePolicy bool
	
	// PolicyName is the name of the PropagationPolicy(or ClusterPropagationPolicy),
	// It defaults to the promoting resource name with a random hash suffix.
	// It will be ignored if AutoCreatePolicy is false.
	PolicyName string
```